File: Daily/PY/Leetcode230-二叉搜索树中第k小的元素.py
class Solution:
    cnt,res = 0,0
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        
        def dfs(root,k):
            if not root or self.cnt > k:
                return
            dfs(root.left,k)
            self.cnt += 1
            if self.cnt == k:
                self.res = root.val
                return 
            dfs(root.right,k)
        dfs(root,k) 

        return self.res

File: Leetcode/sbgo/Leetcode230-二叉搜索树中第k小的元素.go
package sbgo

func kthSmallest(root *TreeNode, k int) int {
	res, cnt := 0, 0
	var dfs func(*TreeNode, int)
	dfs = func(root *TreeNode, k int) {
		if root == nil || cnt > k {
			return
		}
		dfs(root.Left, k)
		cnt++
		if cnt == k {
			res = root.Val
			return
		}
		dfs(root.Right, k)
	}
	dfs(root, k)
	return res
}
